src/connector/src/sink/big_query.rs

```diff
@@ -393,19 +476,188 @@ impl SinkWriter for BigQuerySinkWriter {
     }
 
     async fn barrier(&mut self, _is_checkpoint: bool) -> Result<()> {
-        self.insert_data().await
+        Ok(())
     }
 
     async fn update_vnode_bitmap(&mut self, _vnode_bitmap: Arc<Bitmap>) -> Result<()> {
         Ok(())
     }
 }
 
+struct StorageWriterClient {
+    client: StreamingWriteClient,
+    environment: Environment,
+}
+impl StorageWriterClient {
+    pub async fn new(credentials: CredentialsFile) -> Result<Self> {
+        let ts_grpc = google_cloud_auth::token::DefaultTokenSourceProvider::new_with_credentials(
+            Self::bigquery_grpc_auth_config(),
+            Box::new(credentials),
+        )
+        .await
+        .map_err(|e| SinkError::BigQuery(e.into()))?;
+        let conn_options = ConnectionOptions {
+            connect_timeout: CONNECT_TIMEOUT,
+            timeout: CONNECTION_TIMEOUT,
+        };
+        let environment = Environment::GoogleCloud(Box::new(ts_grpc));
+        let conn = WriteConnectionManager::new(
+            DEFAULT_GRPC_CHANNEL_NUMS,
+            &environment,
+            DOMAIN,
+            &conn_options,
+        )
+        .await
+        .map_err(|e| SinkError::BigQuery(e.into()))?;
+        let client = conn.conn();
+        Ok(StorageWriterClient {
+            client,
+            environment,
+        })
+    }
+
+    pub async fn append_rows(
+        &mut self,
+        rows: Vec<AppendRowsRequestRows>,
+        write_stream: String,
+    ) -> Result<()> {
+        let trace_id = Uuid::new_v4().hyphenated().to_string();
+        let append_req: Vec<AppendRowsRequest> = rows
+            .into_iter()
+            .map(|row| AppendRowsRequest {
+                write_stream: write_stream.clone(),
+                offset: None,
+                trace_id: trace_id.clone(),
+                missing_value_interpretations: HashMap::default(),
+                rows: Some(row),
+            })
+            .collect();
+        let resp = self
+            .client
+            .append_rows(Request::new(tokio_stream::iter(append_req)))
```

Comment:
The resp is a response stream. May need to wait for the response of all requests.